### ISSUE TITLE
dungeon rarity and other questions

### ISSUE THREAD

**Issue by whitewolfxs** [Greymerk/minecraft-roguelike#39]
i was wondering if when u make your own custom dungeon if u can have a rarity system like : this dungeon only spawns half of the time as the normal ones do..also if it is possible for this new dungeons to have their very own specific loot table (this to try make harder duegeons with better loot).
ps : sorry for bad english :3

**Comment by Greymerk** [comment]
Yes, any custom settings will override the defaults. However you can weight your custom settings as part of the criteria.

All custom settings by default have a weight of 1, unless otherwise specified.

should look something like this:

```javascript
{
	"name": "demo",
	"criteria": {
		"weight": 1
	}
}
```

**Comment by whitewolfxs** [comment]
and u can atach a loot file to a specific dungeon?

**Comment by Greymerk** [comment]
Yes, you can alter loot for specific dungeons. In the current build actually the desert dungeon is slightly harder than the grasslands one, and there's extra gold etc added to chests. Mostly for flavor. In jungle dungeons you'll find extra emeralds.

I'll create a sample configuration when I get some time.

**Comment by whitewolfxs** [comment]
ok thank you very much for all the help :)

**Comment by petrukhnov** [comment]
from wiki it is not clear how to get less dungeons... 
Something like: 'use 20 instead of 10 to reduce number of dungeons' will help

**Comment by dshadowwolf** [comment]
FYI: From careful reading of the source, the value of "spawnFrequency" is inverse - the lower the value, the more frequently dungeons will spawn. 

Through actual testing I've found that it is, conversely, almost on a hyperbolic curve as the number increases above 10. At 20 I've explored an area of about 1k square chunks and seen, if memory serves, one dungeon - in the same area at 10 I'd seen over 100. For that same area, at 15, I saw 20 to 30 - though at that point I was more just confirming that they were generating and not generating in huge numbers.

So... Try setting "spawnFrequency" to somewhere between 11 and 15 (inclusive) if you want the dungeons to be rare but not "there are more strongholds than dungeons" rare.
